cider/include/cider/processor/JoinHashTableBuilder.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_JOIN_HASH_TABLE_BUILDER_H
+#define CIDER_JOIN_HASH_TABLE_BUILDER_H
+
+#include <memory>
+#include "include/cider/batch/CiderBatch.h"
+#include "substrait/algebra.pb.h"
+
+namespace cider::processor {
+
+class JoinHashTable;
+
+class JoinHashTableBuilder {
```

Comment:
Should this class has ```const std::shared_ptr<CiderAllocator> allocator_;```?

Reply:
I'm not too sure about how to use this allocator , but yes I will add it

Reply:
fixed